--- /var/www/html/wiki14org/languages/Language.php	2005-03-13 20:13:39.000000000 -0600
+++ /var/www/html/wiki14/languages/Language.php	2005-04-09 17:08:53.124331977 -0500
@@ -519,10 +519,12 @@
 'yourname'		=> 'Your user name',
 'yourpassword'	=> 'Your password',
 'yourpasswordagain' => 'Retype password',
 'newusersonly'	=> ' (new users only)',
 'remembermypassword' => 'Remember my password across sessions.',
+'yourdomainname'       => 'Your domain',
+'externaldberror'      => 'There was either an external authentication database error or you are not allowed to update your external account.',
 'loginproblem'	=> '<b>There has been a problem with your login.</b><br />Try again!',
 'alreadyloggedin' => "<font color=red><b>User $1, you are already logged in!</b></font><br />\n",
 
 'login'			=> 'Log in',
 'loginprompt'           => "You must have cookies enabled to log in to {{SITENAME}}.",
@@ -551,11 +553,11 @@
 'loginsuccesstitle' => 'Login successful',
 'loginsuccess'	=> "You are now logged in to {{SITENAME}} as \"$1\".",
 'nosuchuser'	=> "There is no user by the name \"$1\".
 Check your spelling, or use the form below to create a new user account.",
 'nosuchusershort'	=> "There is no user by the name \"$1\". Check your spelling.",
-'wrongpassword'	=> 'The password you entered is incorrect. Please try again.',
+'wrongpassword'                => 'The password you entered is incorrect, missing, or the authentication database could not be contacted. Please try again.',
 'mailmypassword' => 'Mail me a new password',
 'passwordremindertitle' => "Password reminder from {{SITENAME}}",
 'passwordremindertext' => "Someone (probably you, from IP address $1)
 requested that we send you a new {{SITENAME}} login password.
 The password for user \"$2\" is now \"$3\".
